arviz/plots/distcomparisonplot.py

```diff
@@ -0,0 +1,195 @@
+    transform : callable
+        Function to transform data (defaults to None i.e. the identity function)
+    legend : bool
+        Add legend to figure. By default True.
+    ax: axes, optional
+        Matplotlib axes: The ax argument should have shape (nvars, 3), where the
+        last column is for the combined before/after plots and columns 0 and 1 are
+        for the before and after plots, respectively.
+    prior_kwargs : dicts, optional
+        Additional keywords passed to `arviz.plot_dist` for prior/predictive groups.
+    posterior_kwargs : dicts, optional
+        Additional keywords passed to `arviz.plot_dist` for posterior/predictive groups.
+    observed_kwargs : dicts, optional
+        Additional keywords passed to `arviz.plot_dist` for observed_data group.
+    backend: str, optional
+        Select plotting backend {"matplotlib","bokeh"}. Default "matplotlib".
+    backend_kwargs: bool, optional
+        These are kwargs specific to the backend being used. For additional documentation
+        check the plotting method of the backend.
+    show : bool, optional
+        Call backend show function.
+
+    Returns
+    -------
+    axes : a numpy 2d array of matplotlib axes. Returned object will have shape (nvars, 3),
+    where the last column is the combined plot and the first columns are the single plots.
+
+    Examples
+    --------
+    Plot the prior/posterior plot for specified vars and coords.
+
+    .. plot::
+        :context: close-figs
+
+        >>> import arviz as az
+        >>> data = az.load_arviz_data('radon')
+        >>> az.plot_dist_comparison(data, var_names=["defs"], coords={"team" : ["Italy"]})
+
+    """
+
+    all_groups = ["prior", "posterior"]
+
+    if kind == "observed":
+        all_groups = ["observed_data", "prior_predictive", "posterior_predictive"]
+
+    if coords is None:
+        coords = {}
+
+    if prior_kwargs is None:
+        prior_kwargs = {}
+
+    if posterior_kwargs is None:
+        posterior_kwargs = {}
+
+    if observed_kwargs is None:
+        observed_kwargs = {}
+
+    if backend_kwargs is None:
+        backend_kwargs = {}
+
+    datasets = []
+    groups = []
+    for group in all_groups:
+        try:
+            datasets.append(getattr(data, group))
+            groups.append(group)
+        except:  # pylint: disable=bare-except
+            pass
+
+    if var_names is None:
+        var_names = list(datasets[0].data_vars)
+
+    if isinstance(var_names, str):
+        var_names = [var_names]
+
+    if isinstance(var_names[0], str):
+        var_names = [var_names for _ in datasets]
+
+    var_names = [_var_names(vars, dataset) for vars, dataset in zip(var_names, datasets)]
+
+    if transform is not None:
+        datasets = [transform(dataset) for dataset in datasets]
+
+    datasets = get_coords(datasets, coords)
+    dc_plotters = [
+        list(xarray_var_iter(data, var_names=var, combined=True))
+        for data, var in zip(datasets, var_names)
+    ]
```

Comment:
cut `dc_plotters` list so that no more than `rcParams["plot.max_subplots"]` are generated (also ensure at least one var is plotted, if max_subplots were 2, plot one variable instead of none, plotting variables partially is not worth the effort).